src/Arcus.WebApi.Logging/RequestTrackingMiddleware.cs

```diff
@@ -133,20 +179,38 @@ private async Task<string> GetRequestBodyAsync(HttpContext httpContext)
         {
             if (_options.IncludeRequestBody)
             {
-                _logger.LogTrace("Prepare for the request body to be tracked...");
-                string sanitizedBody = await SanitizeRequestBodyAsync(httpContext.Request.Body);
-                if (sanitizedBody != null)
-                {
-                    _logger.LogTrace("Found request body to be tracked");
-                    return sanitizedBody;
-                }
+                string sanitizedBody = await GetBodyAsync(httpContext.Request.Body, _options.RequestBodyBufferSize, "Request");
+                return sanitizedBody;
+            }
+
+            return null;
+        }
 
-                _logger.LogWarning("No request body was found to be tracked");
+        private async Task<string> GetResponseBodyAsync(HttpContext httpContext)
+        {
+            if (_options.IncludeResponseBody)
+            {
+                string sanitizedBody = await GetBodyAsync(httpContext.Response.Body, _options.ResponseBodyBufferSize, "Response");
+                return sanitizedBody;
             }
 
             return null;
         }
 
+        private async Task<string> GetBodyAsync(Stream body, int? maxLength, string target)
+        {
+            _logger.LogTrace("Prepare for {Target} body to be tracked...", target);
+            string sanitizedBody = await SanitizeStreamAsync(body, maxLength, target);
+            if (sanitizedBody != null)
+            {
+                _logger.LogTrace("Found {Target} body to be tracked", target);
+                return sanitizedBody;
+            }
+                
+            _logger.LogWarning("No {Target} body was found to be tracked", target);
```

Comment:
I'd use a trace, otherwise for OK/NoContent/Accepted responses we will always generate warnings
```suggestion
            _logger.LogTrace("No {Target} body was found to be tracked", target);
```

Reply:
Ah, yes, that would be better indeed! Thanks.